### PR TITLE
Turn off warnings to get ZetaSQL to build.

### PR DESCRIPTION
**Pull request by markww**

### CODE REVIEW

File: .bazelrc [hunk 1]
```diff
@@ -44,7 +49,27 @@ build --cxxopt='-Wno-final-dtor-non-final-class'
...
+build --cxxopt='-Wno-enum-compare-switch'
+build --host_cxxopt='-Wno-enum-compare-switch'
+build --cxxopt='-Wno-char-subscripts'
+build --host_cxxopt='-Wno-char-subscripts'
+# This warning has a false positive on ZetaSQL.
+build --cxxopt='-Wno-thread-safety-precise'
+build --host_cxxopt='-Wno-thread-safety-precise'
 
+# Need this for ZetaSQL's icu dependency as long as it remains at version 65.
+# Should remove when it gets upgraded.
```
Comment (bgogul) [inline .bazelrc:70]: File a bug?

File: .bazelrc [hunk 2]
```diff
@@ -44,7 +49,27 @@ build --cxxopt='-Wno-final-dtor-non-final-class'
 build --host_cxxopt='-Wno-final-dtor-non-final-class'
 build --cxxopt='-Wno-unused-const-variable'
 build --host_cxxopt='-Wno-unused-const-variable'
+build --cxxopt='-Wno-unused-variable'
+build --host_cxxopt='-Wno-unused-variable'
+build --cxxopt='-Wno-unused-function'
+build --host_cxxopt='-Wno-unused-function'
```
Comment (bgogul) [inline .bazelrc:55]: Are these for ZetaSQL as well? If so, move it after the comment on line 56?

**Review by bgogul — APPROVED** [review]

File: .bazelrc (hunk 2 above)
Comment (markww) [reply .bazelrc:55]: There are two sections here: warnings we are turning off for ZetaSQL for which I have proposed fixes commits to the ZetaSQL folks that should address the warnings, and warnings that we are turning off for ZetaSQL where I don't know if a fix will be published soon. These are in the former category, and there's already a header addressing them.

File: .bazelrc (hunk 1 above)
Comment (markww) [reply .bazelrc:70]: https://github.com/google/zetasql/issues/122 .

**Pull request merged**